### PR TITLE
Addonify!

### PR DESCRIPTION
**Pull request by rwjblue**
* convert to an addon
* migrate shims to vendor/ folder
* handle ember-source@2.11 alpha/beta series separately from
  other ember-source versions.

### CODE REVIEW

File: tests/unit/general-importing-test.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import require from 'require';
+import { module, test } from 'qunit';
+
+module('shims generally work');
```
Comment (twokul) [inline tests/unit/general-importing-test.js]: should we test all shim'd modules?

Comment (rwjblue) [inline tests/unit/general-importing-test.js]: Yeah, I thought about it.  Not sure.

I'd have to duplicate the structure which seems weird/odd...

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+---
+language: node_js
+node_js:
+  - "6"
```
Comment (twokul) [inline .travis.yml]: don't care about 4?

**Comment by rwjblue** [comment]
Annoyingly the tests here will fail until the upstream ember-cli changes are landed and released.  Merging this so that I can proceed with that...

**Comment by Turbo87** [comment]
@rwjblue reviewed, seems good :+1:

**Comment by nathanhammond** [comment]
Offline conversation with @rwjblue we discussed our three compatibility options:

- Continue make it work via `bower` by keeping everything at `/app-shims.js`. The scenario that triggers this is a person manually updating their `bower.json` in a pre-2.11 Ember CLI app.
- Add `throw new Error()` to `/app-shims.js` in this version. This will move the error to run time and allow us to provide a good error message. However, there is the chance that this results in somebody shipping a broken thing without knowing it. They'll blame themselves, but we set them up for that failure.
- Keep this exactly as this PR has it. This will fail build time with a incomprehensible error since it won't find a file during a `broccoli-concat`. This results in higher support costs, but users won't accidentally ship a broken application.

We are electing to make no change to how this PR has it.

**Pull request merged**